projects/RabbitMQ.Client/Impl/Channel.PublisherConfirms.cs

```diff
@@ -329,5 +342,49 @@ await publisherConfirmationInfo.MaybeWaitForConfirmationAsync(cancellationToken)
                 }
             }
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private ValueTask MaybeEnforceOutstandingPublisherConfirmationsAsync(CancellationToken cancellationToken)
+        {
+            if (_publisherConfirmationTrackingEnabled)
+            {
+                if (_maxOutstandingPublisherConfirmsReached.IsSet)
+                {
+                    return default;
+                }
+                else
+                {
+                    return _maxOutstandingPublisherConfirmsReached.WaitAsync(cancellationToken);
+                }
+            }
+
+            return default;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private void MaybeBlockPublishers()
+        {
+            if (_publisherConfirmationTrackingEnabled)
+            {
+                if (_maxOutstandingPublisherConfirmations is not null
```

Comment:
Wouldn't it be way simpler to have another semaphore slim that is set to the number of outstanding confirms allowed or not set in the unlimited mode?

Reply:
That makes _WAY MORE SENSE_ 🤦‍♂️